legal-api/src/legal_api/resources/business/business_comments.py

```diff
@@ -0,0 +1,121 @@
+
+from legal_api.exceptions import BusinessException
+from legal_api.models import Business, Comment, User, db
+from legal_api.services import authorized
+from legal_api.services.comments import validate
+from legal_api.utils.auth import jwt
+from legal_api.utils.util import cors_preflight
+
+from .api_namespace import API
+# noqa: I003; the multiple route decorators cause an erroneous error in line space counting
+
+
+@cors_preflight('GET, POST')
+@API.route('/<string:identifier>/comments', methods=['GET', 'POST', 'OPTIONS'])
+@API.route('/<string:identifier>/comments/<int:comment_id>', methods=['GET', 'OPTIONS'])
+class BusinessCommentResource(Resource):
+    """Business Comment service."""
+
+    @staticmethod
+    @cors.crossdomain(origin='*')
+    @jwt.requires_auth
+    def get(identifier, comment_id=None):
+        """Return a JSON object with meta information about the Service."""
+        # basic checks
+        business = Business.find_by_identifier(identifier)
+        err_msg, err_code = BusinessCommentResource._basic_checks(identifier, business, request)
+        if err_msg:
+            return jsonify(err_msg), err_code
+
+        comments = db.session.query(Comment).filter(Comment.business_id == business.id, Comment.filing_id.is_(None))
+
+        if comment_id:
+            comment = comments.filter(Comment.id == comment_id).one_or_none()
+            if not comment:
+                return jsonify({'message': f'Comment {comment_id} not found'}), HTTPStatus.NOT_FOUND
+
+            return jsonify(comment.json)
+
+        rv = []
+        for comment in comments:
+            rv.append(comment.json)
+
+        return jsonify(comments=rv)
+
+    @staticmethod
+    @cors.crossdomain(origin='*')
+    @jwt.requires_auth
+    def post(identifier):
+        """Create a new comment for the business."""
+        # basic checks
+        business = Business.find_by_identifier(identifier)
+        err_msg, err_code = BusinessCommentResource._basic_checks(identifier, business, request)
+        if err_msg:
+            return jsonify(err_msg), err_code
+
+        json_input = request.get_json()
+
+        # check authorization
+        if not authorized(identifier, jwt, action=['add_comment']):
+            return jsonify({'message':
+                            f'You are not authorized to submit a comment for {identifier}.'}), \
+                HTTPStatus.UNAUTHORIZED
+
+        # validate comment
+        err = validate(json_input, False)
+        if err:
+            json_input['errors'] = err.msg
+            return jsonify(json_input), err.code
+
+        # save comment
+        user = User.get_or_create_user_by_jwt(g.jwt_oidc_token_info)
+        try:
+            comment = Comment()
+            comment.comment = json_input['comment']['comment']
+            comment.staff_id = user.id
+            comment.business_id = business.id
+            comment.timestamp = datetime.datetime.utcnow()
+            comment.save()
```

Comment:
Does this reuse the existing comments table?

Where is field `comment.submitterDisplayName`?

Reply:
Yes its existing comments table. `submitterDisplayName` is `display_name` from `user` table, which will be in get response